modules/collection/programs/kitty.nix

```
@@ -0,0 +1,113 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  inherit (lib.generators) mkKeyValueDefault;
+  inherit (lib.lists) hasPrefix;
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+  inherit (lib.strings) typeOf;
+  inherit (lib.types) oneOf nonEmptyStr submodule path;
+
+  keyValueSettings = {
+    listsAsDuplicateKeys = true;
+    mkKeyValue = mkKeyValueDefault {} " ";
+  };
+
+  keyValue = pkgs.formats.keyValue keyValueSettings;
+
+  themeType = oneOf [
+    path
+    nonEmptyStr
+  ];
+  lightDarkType = submodule {
+    options = {
+      light = mkOption {type = themeType;};
+      dark = mkOption {type = themeType;};
+    };
+  };
+  kittyThemeType = oneOf [
+    themeType
+    lightDarkType
+  ];
+
+  mkTheme = theme:
+    if ((typeOf theme) == "string") && !(hasPrefix "/" theme)
+    then "${pkgs.kitty-themes}/share/kitty-themes/themes/${theme}.conf"
+    else theme;
+
+  cfg = config.rum.programs.kitty;
+in {
+  options.rum.programs.kitty = {
+    enable = mkEnableOption "kitty";
+
+    package = mkPackageOption pkgs "kitty" {};
+
+    settings = mkOption {
+      type = keyValue.type;
+      default = {};
+      example = {
+        font_size = 10;
+        text_composition_strategy = "2.5 0";
+        cursor_shape = "beam";
+        map = [
+          "kitty_mod+l next_layout"
+          "ctrl+alt+t goto_layout tall"
+        ];
+      };
+      description = ''
+        The configuration converted and written to {file}`$HOME/.config/kitty/kitty.conf`.
+        Please reference https://sw.kovidgoyal.net/kitty/conf/ for config options.
```

```suggestion
        Please reference [kitty.conf's online reference](https://sw.kovidgoyal.net/kitty/conf/) for config options.
```

Please also add a reference to the [manpage](https://manpages.debian.org/testing/kitty/kitty.conf.5.en.html), and add it into `manpage-urls.json`.

---

Where is the `manpage-uris.json`?

---

Oops, I forgot the documentation PR didn't get merged yet. You can simply reference the manual, with ``{manpage}`kitty.conf(5)` `` and we will take care of adding it there once we get documentation.